src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/LSPIJUtils.java

```diff
@@ -319,15 +319,10 @@ public static void applyEdits(Editor editor, Document document, List<TextEdit> e
         ApplicationManager.getApplication().runWriteAction(() -> edits.forEach(edit -> applyEdit(editor, edit, document)));
     }
 
-    public static boolean hasCapability(Either<Boolean, ? extends Object> eitherCapability) {
-        if(eitherCapability != null) {
-            if (eitherCapability.isLeft()) {
-                return eitherCapability.getLeft();
-            } else {
-                return eitherCapability.getRight() != null;
-            }
-        } else {
+    public static boolean hasCapability(final Either<Boolean, ? extends Object> eitherCapability) {
+        if(eitherCapability == null) {
             return false;
         }
+        return eitherCapability.isRight() || (eitherCapability.isLeft() && eitherCapability.getLeft());
```

Comment:
if you want to be thorough: 
> return (eitherCapability.isRight() && eitherCapability.getRight() != null  &&  eitherCapability.getRight()) || (eitherCapability.isLeft() && eitherCapability.getLeft() != null  &&  eitherCapability.getLeft());

Reply:
I dont think it is correct since getObject is not à boolean. The main idea is to check if left is true or right is filled.

Reply:
OK, fair enough